packages/core-database/src/database-service.ts

```diff
@@ -568,7 +568,8 @@ export class DatabaseService implements Database.IDatabaseService {
         const dbTransaction = await this.getTransaction(transaction.data.id);
 
         try {
-            return transactionHandler.canBeApplied(transaction, sender, this.walletManager) && !dbTransaction;
+            transactionHandler.canBeApplied(transaction, sender, this.walletManager);
```

Comment:
Please revert, this just looks off.

Reply:
The interface of `canBeApplied()` was flawed. In general there are two patterns:

A. Indicate the result via the return value. For example - `true` for success and `false` for failure. In this case the usage involves checking the returned value:
```
if (foo()) {
  code for success
} else {
  code for failure
}
```

B. Indicate the result by throwing an exception on failure and not throwing on success. In this case the return value is irrelevant, and so the function does not return anything (ie `void`). In this case the usage involves try/catch:
```
try {
  foo()
  code for success
} catch () {
  code for failure
}
```

It is arguable whether it is good to throw an exception to indicate an error in a user supplied input and that A should be preferred over B, but that is another topic.

Before this patch we had the interface of `canBeApplied()` be a mixture of A and B - it would throw on failure and return `true` on success. Which is flawed because the return value is redundant - the caller knows that if it returned, then it succeeded and the return value is *always* `true`.

So, please pick one of A or B, but not a mix of both. With this patch I have made it B.

What is this with tslint? Would it insist that every function returns something!? Frowning upon `void` functions in general?

Reply:
Then `canBeApplied` needs at least a different name that doesn't imply it returns a boolean.

Reply:
Right, `canBeApplied` gives the impression that it would return a boolean. What about `checkApplicability` or `checkIfCanBeApplied`? Btw, I would have coded this using variant A - returning boolean true/false and no exceptions to indicate wrong user input, but I kept it with the exceptions because I thought that the author likes the exceptions. Maybe I should change it to A. and keep the name `canBeApplied`?

Reply:
Without exceptions you lose the information about why it failed. Meaning you need to refactor a lot more than just the return value. So I wouldn't change it for the sake of changing it. There are more important issues to work on.

`throwIfCannotBeApplied()`?

Reply:
Renamed.